semgrep-core/src/parsing/tree_sitter/Parse_rust_tree_sitter.ml

```diff
@@ -3211,9 +3211,13 @@ and map_declaration_statement (env : env) (*_outer_attrs _visibility*) x :
             ty)
           v4
       in
-      let _tyTODO = map_type_ env v5 in
+      let ty = map_type_ env v5 in
       let _where_clauseTODO = Option.map (fun x -> map_where_clause env x) v6 in
-      map_decls_or_semi env v7
+      let body = map_decls_or_semi env v7 in
+      (* TODO not sure what to put for the name *)
```

Comment:
not "impl" for sure :)
This is similar to type classes in Haskell. It does not really define an entity, instead it extends the definition
of an existing entity/type (in your example Ptr). We probably need a new construct for that in AST_generic.ml